libr/util/format.c

```diff
@@ -57,17 +57,36 @@ static float updateAddr(const ut8 *buf, int len, int endian, ut64 *addr, ut64 *a
 
 static int r_get_size(RNum *num, ut8 *buf, int endian, const char *s) {
 	int size = 0, len = strlen (s);
-	ut64 addr;
-
 	if (s[0] == '*' && len >= 4) { // value pointed by the address
+		ut64 addr;
 		int offset = (int)r_num_math (num, s + 1);
 		(void)updateAddr (buf + offset, 999, endian, &addr, NULL);
 		return addr;
+	}
+	// flag handling doesnt seems to work here
+	return size = r_num_math (num, s);
+}
+
+static void r_print_format_u128(const RPrint* p, int endian, int mode,
+		const char *setval, ut64 seeki, ut8* buf, int i, int size) {
+	ut64 low = r_read_ble64 (buf, endian);
+	ut64 hig = r_read_ble64 (buf + 8, endian);
+	if (MUSTSEEJSON) {
+		p->cb_printf ("\"");
+	} else if (!SEEVALUE && !ISQUIET) {
+		p->cb_printf ("0x%08"PFMT64x" = (uint128_t)", seeki);
+	}
+	if (endian) {
+		p->cb_printf ("0x%016"PFMT64x"", low);
+		p->cb_printf ("%016"PFMT64x, hig);
 	} else {
-		// flag handling doesnt seems to work here
-		size = r_num_math (num, s);
+		p->cb_printf ("0x%016"PFMT64x"", hig);
+		p->cb_printf ("%016"PFMT64x, low);
+	}
+	if (MUSTSEEJSON) {
+		const char *end = endian? "big": "little";
+		p->cb_printf ("\",\"endian\":\"%s\",\"ctype\":\"uint128_t\"}", end);
```

Comment:
Why this? I think `q` is behaving differently.
```
pfj Qq~{}
[
  {
    "type": "Q",
    "offset": 26832,
    "value": "0xe48348e289485ed18949ed31fa1e0ff3",
    "endian": "little",
    "ctype": "uint128_t"
  },
  {
    "type": "q",
    "offset": 26848,
    "value": 2028314786606108912
  }
]
```

Reply:
yes i think all types shuold give the ctype and endian fields too i added only here, buit i was planning to add the rest in another pr